src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/BjorhusImpl.cpp

```diff
@@ -0,0 +1,97 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/BjorhusImpl.hpp"
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/LeviCivitaIterator.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace GeneralizedHarmonic::BoundaryConditions::Bjorhus {
+template <size_t VolumeDim, typename DataType>
+void add_constraint_preserving_terms_to_dt_v_zero(
+    const gsl::not_null<tnsr::iaa<DataType, VolumeDim, Frame::Inertial>*>
+        bc_dt_v_zero,
+    const tnsr::I<DataType, VolumeDim, Frame::Inertial>&
+        unit_interface_normal_vector,
+    const tnsr::iaa<DataType, VolumeDim, Frame::Inertial>&
+        four_index_constraint,
+    const std::array<DataType, 4>& char_speeds) noexcept {
+  if (UNLIKELY(get_size(get<0, 0, 0>(*bc_dt_v_zero)) !=
+               get_size(get<0>(unit_interface_normal_vector)))) {
+    *bc_dt_v_zero =
```

Comment:
since the function is named `add`, this shouldn't be getting setting to zero. Can you add a test that if the `bc_dt_v_zero` is `1.0` on input that the output is `1.0 + correction`?

Reply:
As for the same question [here](https://github.com/sxs-collaboration/spectre/pull/2998#discussion_r636060001), I have changed the name and function behavior to not add but simply assign the RHS correction to the returned pointer memory.